Backend/Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM golang:1.23.1-bookworm AS build-stage
+
+WORKDIR /app
+
+COPY go.mod ./
+
+RUN go mod download
+
+COPY . .
+
+RUN CGO_ENABLED=0 GOOS=linux go build -o /app/bin/main ./cmd/main
+
+FROM build-stage AS test-stage
+RUN go test -v ./...
+
+FROM gcr.io/distroless/base-debian11 AS release-stage
+
+WORKDIR /
+
+COPY --from=build-stage /app/bin/main /app/bin/main
+
+EXPOSE 8080
+
+USER nonroot:nonroot
```

Comment:
My cluster thanks you- meant to mention this to you, non-root is always better if it's possible.